internal/k6runner/k6runner.go

```diff
@@ -22,9 +22,24 @@ import (
 	"github.com/spf13/afero"
 )
 
+// Script is a k6 script that a runner is able to run, with some added instructions for that runner to act on.
+type Script struct {
+	Script   []byte   `json:"script"`
+	Settings Settings `json:"settings"`
+	// TODO: Add Metadata and Features.
+}
+
+type Settings struct {
+	// Timeout for k6 run, in milliseconds. This value is a configuration value for remote runners, which will instruct
+	// them to return an error if the operation takes longer than this time to complete. Clients should expect that
+	// requests to remote runners may take longer than this value due to network and other latencies, and thus clients
+	// must not cancel outgoing requests after this time passes. Outgoing requests should be cancel on RetryOn instead.
+	Timeout int64 `json:"timeout"`
+}
+
 type Runner interface {
 	WithLogger(logger *zerolog.Logger) Runner
-	Run(ctx context.Context, script []byte) (*RunResponse, error)
+	Run(ctx context.Context, script Script) (*RunResponse, error)
```

Comment:
This is the core change of this PR: Replace the `[]byte` argument in `Runner.Run` with the new `Script` type. All other changes are essentially adapting the rest of the codebase to this change.